CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+option(SANITIZE_ADDRESS "Address Sanitizer." OFF)
+
+if (SANITIZE_ADDRESS)
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+endif()
+
+include(FetchContent)
+
+if(NOT InstallVendor)
+function(InstallVendor project_name git_repo branch)
+        IF (NOT ${project_name})
+                FetchContent_Declare(
+                        ${project_name}
+                        GIT_REPOSITORY ${git_repo}
+                        GIT_TAG ${branch}
+                        SOURCE_DIR "${PROJECT_SOURCE_DIR}/vendor/${project_name}"
+                        BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}"
+                        SUBBUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}-sub"
+                )
+                FetchContent_MakeAvailable(${project_name})
+                set(${project_name} CACHE INTERNAL "")
+                message(STATUS "Found ${project_name} @ ${branch}")
+        endif()
+endfunction()
+endif()
+
+set(This oof)
+project(${This} LANGUAGES C CXX)
```

Comment:
A header-only project doesn't need any languages enabled at top-level, because it doesn't build anything, so it should be set to `NONE`.

Reply:
At the moment it has no header file but as you have seen we might need some impl C files which make it easier for the project setup in the end. Since we are required to set a Language I am not going to set it to something that it isn't.

Reply:
You can enable languages in subdirectories just fine. Check [this project](https://github.com/friendlyanon/cmake-init-header-only).